app/lib/search/token_index.dart

```diff
@@ -142,6 +143,30 @@ extension StringTokenIndexExt on TokenIndex<String> {
   }
 }
 
+/// A reusable pool for [IndexedScore] instances to spare some memory allocation.
+class ScorePool<K> {
+  final List<K> _keys;
+  final _pool = <IndexedScore<K>>[];
+
+  ScorePool(this._keys);
+
+  R withScore<R>({
+    required double value,
+    required R Function(IndexedScore<K> score) fn,
+  }) {
+    late IndexedScore<K> score;
+    if (_pool.isNotEmpty) {
+      score = _pool.removeLast();
+      score._values.setAll(0, Iterable.generate(score.length, (_) => value));
+    } else {
+      score = IndexedScore<K>(_keys, value);
+    }
+    final r = fn(score);
+    _pool.add(score);
```

Comment:
This logic means that we will never have more scores then the high-watermark of scores that we need at one time - right?

Thus we don't really risk leaking excessive memory... right?

Reply:
That's correct. We also replace the index every 15 minutes, and if there is an extreme outlier period, its memory footprint won't be around after that.